tests/unittest_brain.py

```diff
@@ -106,14 +110,14 @@ def _assert_hashlib_class(self, class_obj):
         self.assertEqual(len(class_obj["digest"].args.args), 1)
         self.assertEqual(len(class_obj["hexdigest"].args.args), 1)
 
-    def test_hashlib(self):
+    def test_hashlib(self) -> None:
```

Comment:
I'm not really sure we should strictly type all tests. There isn't really much of a benefit in doing so. Those are just tests after all. Especially adding `-> None` to every test method seems unnecessary. IMO we shouldn't do that.

--
I do however agree that typing test helper methods will be quite useful. (Mostly for improved intellisense.)

--
Just FYI: Home Assistant only runs `flake` and `black` over the test files. `mypy` and `pylint` only for the production code.

Reply:
I think once it's done having it is better than not having it. Also useful to have mypy on the whole codebase (it's easier to not have a specific config for tests).

Reply:
> I think once it's done having it is better than not having it. 

Don't believe you can convince me on that one 😄

> Also useful to have mypy on the whole codebase (it's easier to not have a specific config for tests)

Excluding the `tests` folder is pretty easy.

--
Anyway, the topic is not worth the arguments. Just do what you think.